torchtitan/experiments/qwen3/model/state_dict_adapter.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""
+This script is adapted from torchtitan/models/llama3/model/state_dict_adapter.py.
+
+We can use this script to adapt the checkpoint from HF to the format that we can load into the torchtitan model and vice versa.
+This can enable us to do a parity test with the HF implementation and make sure that our results are
+aligned with the HF implementation.
+
+"""
+import re
+from typing import Any
+
+from torchtitan.protocols.state_dict_adapter import StateDictAdapter
+
+from .args import Qwen3ModelArgs
+
+
+class Qwen3StateDictAdapter(StateDictAdapter):
+    def __init__(self, model_args: Qwen3ModelArgs):
```

Comment:
Need to rebase on to main branch, now the StateDictAdapter.__Init__ takes 3 parameters, with hf_asset_path